### PR TITLE
Add product type to typescript types

### PR DESCRIPTION
**Pull request by vegaro**
We are sending `productType` in the Kotlin and Swift side, but the property is missing from the types

### CODE REVIEW

File: typescript/apitesters/offerings.ts [hunk 1]
```diff
@@ -22,6 +22,7 @@ function checkProduct(product: PurchasesStoreProduct) {
   const discounts: PurchasesStoreProductDiscount[] | null = product.discounts;
   const subscriptionPeriod: string | null = product.subscriptionPeriod;
   const productCategory: PRODUCT_CATEGORY | null = product.productCategory;
```
Comment (vegaro) [inline typescript/apitesters/offerings.ts:24]: I am not sure why this is nullable 🤔

File: typescript/apitesters/offerings.ts [hunk 2]
```diff
@@ -126,43 +127,56 @@ function checkPrice(price: Price) {
...
+function checkOfferProductType(productType: PRODUCT_TYPE) {
+  switch (productType) {
+    case PRODUCT_TYPE.CONSUMABLE:
+    case PRODUCT_TYPE.NON_CONSUMABLE:
+    case PRODUCT_TYPE.NON_RENEWABLE_SUBSCRIPTION:
+    case PRODUCT_TYPE.AUTO_RENEWABLE_SUBSCRIPTION:
+    case PRODUCT_TYPE.PREPAID_SUBSCRIPTION:
+    case PRODUCT_TYPE.UNKNOWN:
+      break;
+
```
Comment (tonidero) [inline typescript/apitesters/offerings.ts:180]: Extra line break
```suggestion
```

File: typescript/src/offerings.ts [hunk 3]
```diff
@@ -107,6 +107,10 @@ export interface PurchasesStoreProduct {
      * Product category.
      */
     readonly productCategory: PRODUCT_CATEGORY | null;
+    /**
+     * Product type.
```
Comment (tonidero) [inline typescript/src/offerings.ts:111]: Hmm should we improve the documentation while we're at it?

File: typescript/src/offerings.ts [hunk 4]
```diff
@@ -135,12 +139,44 @@ export enum PRODUCT_CATEGORY {
...
+     */
+    NON_RENEWABLE_SUBSCRIPTION = "NON_RENEWABLE_SUBSCRIPTION",
+
+    /**
+     * An auto-renewable subscription.
+     */
+    AUTO_RENEWABLE_SUBSCRIPTION = "AUTO_RENEWABLE_SUBSCRIPTION",
+
+    /**
+     * A subscription that is pre-paid.
```
Comment (tonidero) [inline typescript/src/offerings.ts:176]: Correct me if I'm wrong but are some of these iOS/Android only? If so, should we add it to the documentation?

**Review by tonidero — APPROVED** [review]

File: typescript/src/offerings.ts (hunk 4 above)
Comment (vegaro) [reply typescript/src/offerings.ts:176]: That's correct, I will add it to the docs

**Review by tonidero — APPROVED** [review]

File: typescript/src/offerings.ts [hunk 5]
```diff
@@ -107,6 +107,10 @@ export interface PurchasesStoreProduct {
      * Product category.
      */
     readonly productCategory: PRODUCT_CATEGORY | null;
+    /**
+     * The specific type of subscription or one time purchase this product represents
```
Comment (NachoSoto) [inline typescript/src/offerings.ts:111]: Can you add a warning to indicate that this isn't available for StoreKit 1 products? https://github.com/RevenueCat/purchases-ios/blob/4.26.1/Sources/Purchasing/StoreKitAbstractions/StoreProduct.swift#L111

Comment (tonidero) [reply typescript/src/offerings.ts:111]: Did this in #515

**Pull request merged**